tesseract_kinematics/kdl/include/tesseract_kinematics/kdl/kdl_utils.h

```diff
@@ -102,6 +102,22 @@ struct KDLChainData
   std::vector<std::pair<std::string, std::string>> chains; /**< The chains used to create the object */
 };
 
+/**
+ * @brief The KDLConfig struct
+ *
+ * This contains parameters that can be used to customize the KDL solver for your application.
+ * They are ultimately passed to the constuctor of the undelying ChainIkSolver.
+ *
+ * The defaults provided here are the same defaults imposed by the KDL library.
+ */
+struct KDLConfig
+{
+  Eigen::VectorXd weights = Eigen::VectorXd::Constant(6, 1.0);
```

Comment:
The default assumes 6 joint which is not good. It may be better to unit size as the default and add a check in the construction to handle this so it creates the correct size based on the number of actual joints.

Reply:
I think these may be task space coordinate weights (i.e., [x, y, z, rx, ry, rz]). It looks like [the default values are actually [1, 1, 1, 0.01, 0.01, 0.01]](https://docs.ros.org/en/kinetic/api/orocos_kdl/html/classKDL_1_1ChainIkSolverPos__LMA.html#a1e83b8d5bb692211d4216e491253afa4), which seems like position only IK. @seanmc602 is this correct?

Reply:
Yes, task space weights. I'll update the variable name to match the config name to be more clear.